rules/get_genome.smk

```diff
@@ -17,17 +17,90 @@ rule get_genome:
         parallel_downloads=1
     priority: 1
     params:
-        config['genome_dir']
+        dir=config['genome_dir'],
+        gtf=expand("{genome_dir}/{{assembly}}/{{assembly}}.annotation.gtf.gz", **config),
     conda:
         "../envs/get_genome.yaml"
     shell:
         """
         active_plugins=$(genomepy config show | grep -Po '(?<=- ).*' | paste -s -d, -)
-        trap "genomepy plugin enable {{$active_plugins}} > {log} 2>&1" EXIT
+        trap "genomepy plugin enable {{$active_plugins,}} > {log} 2>&1" EXIT
 
         genomepy plugin disable {{blacklist,bowtie2,bwa,gaps,gmap,hisat2,minimap2}} >> {log} 2>&1
 
-        genomepy install --genome_dir {params} {wildcards.assembly} UCSC    >> {log} 2>&1 ||
-        genomepy install --genome_dir {params} {wildcards.assembly} NCBI    >> {log} 2>&1 ||
-        genomepy install --genome_dir {params} {wildcards.assembly} Ensembl >> {log} 2>&1
+        genomepy install --genome_dir {params.dir} {wildcards.assembly} UCSC --annotation    >> {log} 2>&1 ||
+        genomepy install --genome_dir {params.dir} {wildcards.assembly} NCBI --annotation    >> {log} 2>&1 ||
+        genomepy install --genome_dir {params.dir} {wildcards.assembly} Ensembl --annotation >> {log} 2>&1
+        
+        gunzip {params.gtf} >> {log} 2>&1
         """
+
+
+rule get_transcripts:
+    """
+    Combine genome.fasta and annotation.gtf to create transcripts.fasta using gffread.
+
+    Prepare genome.fasta if needed.
+    """
+    input:
+        fa=expand("{genome_dir}/{{assembly}}/{{assembly}}.fa", **config),
+        gtf=expand("{genome_dir}/{{assembly}}/{{assembly}}.annotation.gtf", **config)
+    output:
+        expand("{genome_dir}/{{assembly}}/{{assembly}}.transcripts.fa", **config)
+    log:
+        expand("{log_dir}/get_genome/{{assembly}}.transcripts.log", **config)
+    benchmark:
+        expand("{benchmark_dir}/get_genome/{{assembly}}.transcripts.benchmark.txt", **config)[0]
+    priority: 1
+    params:
+        path=conda_path("../../envs/get_genome.yaml"),
+        purgedfa=expand("{genome_dir}/{{assembly}}/{{assembly}}.purged.fa", **config)
```

Comment:
Again why is this a parameter? It seems like it is output here. If it is 'nonsense' output you can always call temp() on the output and it will be deleted when not required anymore.

Reply:
file is generated only when input fasta has dirty formatting